src/v/kafka/connection_context.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+#include "kafka/requests/request_context.h"
+#include "kafka/requests/response.h"
+#include "rpc/server.h"
+#include "seastarx.h"
+#include "utils/hdr_hist.h"
+#include "utils/named_type.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/semaphore.hh>
+#include <seastar/core/shared_ptr.hh>
+
+#include <absl/container/flat_hash_map.h>
+
+#include <memory>
+
+namespace kafka {
+
+class protocol;
+
+class connection_context final
+  : public ss::enable_lw_shared_from_this<connection_context> {
+public:
+    struct session_resources {
```

Comment:
nit: now that this is a top level class, i think the session_resources should be commented. I think a session is really the `life of one request` which is not really a session. More like a `request_lifetime`

Reply:
taking another look its only used to pass some state between a couple private methods. i think it can be made private, or even be moved into the .cc file.